### PR TITLE
Configure self-hosted to test the specified image

### PR DESCRIPTION
**Pull request by chadwhitacre**
We weren't testing the image passed in via `image_url`.

Generalization of the fast fix in https://github.com/getsentry/sentry/pull/42953.

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -39,6 +42,12 @@ runs:
         until docker pull "$IMAGE_URL" 2>/dev/null; do
            sleep 10
         done
+    - name: Configure to use the test image
+      shell: bash
+      run: |
+        pwd
```
Comment (hubertdeng123) [inline action.yml]: What is the reason to include `pwd` and `ls -FGl` here?

Comment (chadwhitacre) [reply action.yml]: Debugging. I'm not sure what directory we're in here, we have two cases above based on whether we're checking out self-hosted or already in self-hosted.

**Comment by chadwhitacre** [comment]
Validating in https://github.com/getsentry/sentry/pull/42959.

**Comment by chadwhitacre** [comment]
Here's [a run](https://github.com/getsentry/sentry/actions/runs/3875880798/jobs/6609026917) ...

**Comment by chadwhitacre** [comment]
I don't understand why it's not failing since `image_var` is not present (right?) but should be required.

**Comment by chadwhitacre** [comment]
Yeah I don't get it. I get an empty string for the value of `image_var` when not provided. I would expect a validation error way earlier in the run.

```
  echo "=us.gcr.io/sentryio/sentry:0eba2828f044e750749a8daffdb6c58ac8bf98ca" > .env.custom
```

https://github.com/getsentry/sentry/actions/runs/3875880798/jobs/6609604022#step:5:483

**Comment by chadwhitacre** [comment]
Let's depend on the `project_name`, and leave the status quo as-is for other vars.

**Review by hubertdeng123 — APPROVED** [review]
Seems fine to me, although a bit messy. Too bad requiring the input doesn't seem to be working

**Comment by chadwhitacre** [comment]
Looks good! 👍

```
SENTRY_IMAGE=us.gcr.io/sentryio/sentry:3adb839e6b8dcf90be7873100dfa3d9ce2de63dd
```

https://github.com/getsentry/sentry/actions/runs/3876881234/jobs/6611258501#step:4:496

**Comment by chadwhitacre** [comment]
Quick post-mortem/retro on this w/ @ethanhs ... lesson learned for next time: move to PR/review workflow off the bat in the action vs. direct-to-main in action (and counting on review from the PR in the other repo).

**Pull request merged**